src/utils.ts

```diff
@@ -379,11 +379,9 @@ export function getSdkLoadCount() {
 export async function awaitSdkEvent(eventName: string, predicate?: Action<any>) {
   return await new Promise(resolve => {
     OneSignal.emitter.once(eventName, (event: Event) => {
-      if (predicate) {
-        const predicateResult = predicate(event);
-        if (predicateResult)
-          resolve(event);
-      } else resolve(event);
+      if (!predicate) {
```

Comment:
This seems like a meaningful semantic shift -- what's the intent here?

Reply:
See the commit for [more details](https://github.com/OneSignal/OneSignal-Website-SDK/commit/ed16f5710b57a268401ae29ae59b1f4bca08cf3b)

It looks like `predicateResult` is always of type `void` anyway.

Reply:
Makes sense, but it looks like we're no longer called predicate when present.  Is that avoidable?

Reply:
Decided on removing predicate altogether....

```
export async function awaitSdkEvent(eventName: string) {
  return await new Promise(resolve => {
    OneSignal.emitter.once(eventName, (event: Event) => {
      resolve(event);
    });
  });
}
```

This function is not used very much and it never actually receives a `predicate`.